pybamm/solvers/processed_variable.py

```diff
@@ -645,15 +645,15 @@ def __call__(self, input):
         if isinstance(first_dim, np.ndarray) and isinstance(second_dim, np.ndarray):
             first_dim = first_dim[:, 0, 0]
             second_dim = second_dim[:, 0]
-            return self.interpolant(second_dim, first_dim)
+            return self.interpolant((first_dim, second_dim))
```

Comment:
Thanks, it might be simpler to just replace `interp2d(x, y, z, kind='linear', ...)` with RegularGridInterpolator`((x, y), z.T, method='linear', ...)` rather than swapping around all the arguments in `__call__`. Also, do you need to transpose the result?

Reply:
`interp2d` also shows up on line 231

Reply:
Thanks! I made the requested changes